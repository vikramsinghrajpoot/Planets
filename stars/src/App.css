.App {
  text-align: center;
}

*{
  box-sizing: border-box
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.content{
  /* height: 100%;
  width: 50%; */
  /* background-color: #61dafb; */
  /* padding-top: 100px;*/
  margin-top: -20px;
  margin-left: 20%; 
  padding-left: 100px
}

.paggingButtons{
margin: auto;
width: 400px;
height: 40px;
display: flex;
}
.next-prevButtons{
  flex:1;
}
