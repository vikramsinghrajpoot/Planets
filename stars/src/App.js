import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import BubbleChart from '@weknow/react-bubble-chart-d3';

class App extends Component {

  constructor(props) {
    super(props)
    this.state = {
      planet: []
    }
    this.url = "https://swapi.co/api/planets/"
    this.currentPage = 0
    this.isLoading = false

    this.data = [
      { label: 'CRM', value: 1.3 },
      { label: 'API', value: 1 },
      { label: 'Data', value: 1 },
      { label: 'Commerce', value: 1 },
      { label: 'AI', value: 3 },
      { label: 'Management', value: 5 },
      { label: 'Testing', value: 6 },
      { label: 'Mobile', value: 9 },
      { label: 'Conversion', value: 9 },
      { label: 'Misc', value: 21 },
      { label: 'Databases', value: 22 },
      { label: 'DevOps', value: 22 },
      { label: 'Javascript', value: 23 },
      { label: 'Languages / Frameworks', value: 25 },
      { label: 'Front End', value: 26 },
      { label: 'Content', value: 26 },
    ]
  }

  render() {
    console.log(this.state.planet)

    if (this.state.planet) {
      return (<div > <div className = "content"><BubbleChart
        graph={{
          zoom: 1.1,
          offsetX: -0.05,
          offsetY: -0.01,
        }}
        width={1000}
        height={800}
        showLegend={true} // optional value, pass false to disable the legend.
        legendPercentage={22} // number that represent the % of with that legend going to use.
        legendFont={{
          family: 'Arial',
          size: 12,
          color: '#000',
          weight: 'bold',
        }}
        valueFont={{
          family: 'Arial',
          size: 12,
          color: '#fff',
          weight: 'bold',
        }}
        labelFont={{
          family: 'Arial',
          size: 16,
          color: '#fff',
          weight: 'bold',
        }}
        //Custom bubble/legend click functions such as searching using the label, redirecting to other page
        bubbleClickFunc={this.bubbleClick}
        legendClickFun={this.legendClick}
        // data={this.data}
        data={this.state.planet[this.currentPage]}
      /></div>
      <br/>
       <div className="paggingButtons" >
      <button onClick={this.prevPress.bind(this)} className="next-prevButtons">Prev</button>
      <button onClick={this.nextPress.bind(this)} className="next-prevButtons">Next</button>
    </div>
      </div>)
    } else {
      return (<h1>No data</h1>)
    }

  }

  prevPress() {
    if(this.isLoading || this.currentPage == 0){return}
    this.currentPage = this.currentPage - 1
    this.forceUpdate()

  }

  nextPress() {
      if(this.isLoading || this.currentPage >=3){return}
      this.currentPage = this.currentPage + 1
      this.loadData()
  }


  componentDidMount() {
    this.loadData()
  }

  loadData(){
    this.isLoading = true
    fetch(this.url, {
      method: "get",
      Headers: {
        'Content-Type': 'application/json',
        'Accept': 'application/json',
      }
    })
      .then(response => response.json())
      .then(data => {
        console.log(data.results);
        this.formateData(data)
        this.isLoading = false
      });
  }

  formateData(data) {
    debugger;
    this.url = data.next

    let localPlanet = []
    data.results.map((data, index) => {

      let diameter = parseInt(data.diameter == "unknown" ? "1" : parseInt(data.diameter)/ 1000 )
      if (diameter > 100) {
        diameter = diameter / 100
      }
      const planet = {
        label: data.name,
        value: diameter
      }
      localPlanet.push(planet)
    })
    console.log("Planet:", this.state.planet)
    this.setState(prev=>({
      planet: [...prev.planet , localPlanet]
    }))
  }
}

export default App;
