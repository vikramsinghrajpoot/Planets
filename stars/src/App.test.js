import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Enzyme, {shallow, mount} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import ReactSixteenAdapter from 'react'

Enzyme.configure({adapter:new Adapter()})

describe("App",()=>{
    test("render test",()=>{
      const wrapper = shallow(<App/>)
      const data = [
        { label: 'CRM', value: 1.3 },
        { label: 'API', value: 1 }]
      wrapper.setState({planets:data})
      expect(wrapper.exists()).toBe(true)
    });
});


it('should render correctly with no props', () => {
  const component = shallow(<App/>);
  expect(component).toMatchSnapshot();
});

test("prev test", () =>{
  const component = new App()
   component.prevPress()
})

test("next test", () =>{
  const component = new App()
   component.nextPress()
})

test("test class name",()=>{
  const wrapper = shallow(<App/>);
  expect(wrapper.find('.content')).toBeDefined();
})

it("load undefined data test", async function() {
  const app = new App()
  const response = await app.loadData()
  expect(response).toBeUndefined
});

it("load defined data test", async () => {
  const app = new App()
   const response =  await app.loadData()
    expect(response).toBeDefined()
  
});



  
