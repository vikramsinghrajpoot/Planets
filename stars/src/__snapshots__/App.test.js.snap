// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with no props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        " ",
        <div
          className="content"
        >
          <BubbleChart
            bubbleClickFun={[Function]}
            graph={
              Object {
                "offsetX": -0.05,
                "offsetY": -0.01,
                "zoom": 1.1,
              }
            }
            height={800}
            labelFont={
              Object {
                "color": "#fff",
                "family": "Arial",
                "size": 16,
                "weight": "bold",
              }
            }
            legendClickFun={[Function]}
            legendFont={
              Object {
                "color": "#000",
                "family": "Arial",
                "size": 12,
                "weight": "bold",
              }
            }
            legendPercentage={22}
            showLegend={true}
            valueFont={
              Object {
                "color": "#fff",
                "family": "Arial",
                "size": 12,
                "weight": "bold",
              }
            }
            width={1000}
          />
        </div>,
        <br />,
        <div
          className="paggingButtons"
        >
          <button
            className="next-prevButtons"
            onClick={[Function]}
          >
            Prev
          </button>
          <button
            className="next-prevButtons"
            onClick={[Function]}
          >
            Next
          </button>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      " ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <BubbleChart
            bubbleClickFun={[Function]}
            graph={
              Object {
                "offsetX": -0.05,
                "offsetY": -0.01,
                "zoom": 1.1,
              }
            }
            height={800}
            labelFont={
              Object {
                "color": "#fff",
                "family": "Arial",
                "size": 16,
                "weight": "bold",
              }
            }
            legendClickFun={[Function]}
            legendFont={
              Object {
                "color": "#000",
                "family": "Arial",
                "size": 12,
                "weight": "bold",
              }
            }
            legendPercentage={22}
            showLegend={true}
            valueFont={
              Object {
                "color": "#fff",
                "family": "Arial",
                "size": 12,
                "weight": "bold",
              }
            }
            width={1000}
          />,
          "className": "content",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bubbleClickFun": [Function],
            "bubbleClickFunc": undefined,
            "data": undefined,
            "graph": Object {
              "offsetX": -0.05,
              "offsetY": -0.01,
              "zoom": 1.1,
            },
            "height": 800,
            "labelFont": Object {
              "color": "#fff",
              "family": "Arial",
              "size": 16,
              "weight": "bold",
            },
            "legendClickFun": [Function],
            "legendFont": Object {
              "color": "#000",
              "family": "Arial",
              "size": 12,
              "weight": "bold",
            },
            "legendPercentage": 22,
            "showLegend": true,
            "valueFont": Object {
              "color": "#fff",
              "family": "Arial",
              "size": 12,
              "weight": "bold",
            },
            "width": 1000,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="next-prevButtons"
              onClick={[Function]}
            >
              Prev
            </button>,
            <button
              className="next-prevButtons"
              onClick={[Function]}
            >
              Next
            </button>,
          ],
          "className": "paggingButtons",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Prev",
              "className": "next-prevButtons",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Prev",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Next",
              "className": "next-prevButtons",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Next",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          " ",
          <div
            className="content"
          >
            <BubbleChart
              bubbleClickFun={[Function]}
              graph={
                Object {
                  "offsetX": -0.05,
                  "offsetY": -0.01,
                  "zoom": 1.1,
                }
              }
              height={800}
              labelFont={
                Object {
                  "color": "#fff",
                  "family": "Arial",
                  "size": 16,
                  "weight": "bold",
                }
              }
              legendClickFun={[Function]}
              legendFont={
                Object {
                  "color": "#000",
                  "family": "Arial",
                  "size": 12,
                  "weight": "bold",
                }
              }
              legendPercentage={22}
              showLegend={true}
              valueFont={
                Object {
                  "color": "#fff",
                  "family": "Arial",
                  "size": 12,
                  "weight": "bold",
                }
              }
              width={1000}
            />
          </div>,
          <br />,
          <div
            className="paggingButtons"
          >
            <button
              className="next-prevButtons"
              onClick={[Function]}
            >
              Prev
            </button>
            <button
              className="next-prevButtons"
              onClick={[Function]}
            >
              Next
            </button>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <BubbleChart
              bubbleClickFun={[Function]}
              graph={
                Object {
                  "offsetX": -0.05,
                  "offsetY": -0.01,
                  "zoom": 1.1,
                }
              }
              height={800}
              labelFont={
                Object {
                  "color": "#fff",
                  "family": "Arial",
                  "size": 16,
                  "weight": "bold",
                }
              }
              legendClickFun={[Function]}
              legendFont={
                Object {
                  "color": "#000",
                  "family": "Arial",
                  "size": 12,
                  "weight": "bold",
                }
              }
              legendPercentage={22}
              showLegend={true}
              valueFont={
                Object {
                  "color": "#fff",
                  "family": "Arial",
                  "size": 12,
                  "weight": "bold",
                }
              }
              width={1000}
            />,
            "className": "content",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bubbleClickFun": [Function],
              "bubbleClickFunc": undefined,
              "data": undefined,
              "graph": Object {
                "offsetX": -0.05,
                "offsetY": -0.01,
                "zoom": 1.1,
              },
              "height": 800,
              "labelFont": Object {
                "color": "#fff",
                "family": "Arial",
                "size": 16,
                "weight": "bold",
              },
              "legendClickFun": [Function],
              "legendFont": Object {
                "color": "#000",
                "family": "Arial",
                "size": 12,
                "weight": "bold",
              },
              "legendPercentage": 22,
              "showLegend": true,
              "valueFont": Object {
                "color": "#fff",
                "family": "Arial",
                "size": 12,
                "weight": "bold",
              },
              "width": 1000,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="next-prevButtons"
                onClick={[Function]}
              >
                Prev
              </button>,
              <button
                className="next-prevButtons"
                onClick={[Function]}
              >
                Next
              </button>,
            ],
            "className": "paggingButtons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Prev",
                "className": "next-prevButtons",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Prev",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Next",
                "className": "next-prevButtons",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Next",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
